specification/communication/data-plane/Microsoft.CommunicationServicesAdministration/stable/2021-03-07/phonenumbers.json

```diff
@@ -0,0 +1,951 @@
+            "name": "PhoneNumberAssignmentType",
+            "modelAsString": true
+          }
+        },
+        "capabilities": {
+          "$ref": "#/definitions/PhoneNumberCapabilities"
+        },
+        "cost": {
+          "$ref": "#/definitions/PhoneNumberCost"
+        },
+        "searchExpiresBy": {
+          "format": "date-time",
+          "description": "The date that this search result expires and phone numbers are no longer on hold. A search result expires in less than 15min, e.g. 2020-11-19T16:31:49.048Z.",
+          "type": "string"
+        }
+      }
+    },
+    "PhoneNumberPurchaseRequest": {
+      "description": "The phone number search purchase request.",
+      "type": "object",
+      "properties": {
+        "searchId": {
+          "description": "The search id.",
+          "type": "string"
+        }
+      }
+    },
+    "PhoneNumberOperation": {
+      "description": "Long running operation.",
+      "required": [
+        "createdDateTime",
+        "id",
+        "operationType",
+        "status"
+      ],
+      "type": "object",
+      "properties": {
+        "status": {
+          "description": "Status of operation.",
+          "enum": [
+            "notStarted",
+            "running",
+            "succeeded",
+            "failed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PhoneNumberOperationStatus",
+            "modelAsString": true
+          }
+        },
+        "resourceLocation": {
+          "description": "URL for retrieving the result of the operation, if any.",
+          "type": "string"
+        },
+        "createdDateTime": {
+          "format": "date-time",
+          "description": "The date that the operation was created.",
+          "type": "string"
+        },
+        "error": {
+          "$ref": "../../../Microsoft.CommunicationServicesCommon/preview/2020-11-19-preview1/common.json#/definitions/CommunicationError"
+        },
+        "id": {
+          "description": "Id of operation.",
+          "type": "string"
+        },
+        "operationType": {
+          "description": "The type of operation, e.g. Search",
+          "enum": [
+            "purchase",
+            "releasePhoneNumber",
+            "search",
+            "updatePhoneNumberCapabilities"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PhoneNumberOperationType",
+            "modelAsString": true
+          }
+        },
+        "lastActionDateTime": {
+          "format": "date-time",
+          "description": "The most recent date that the operation was changed.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "AcquiredPhoneNumber": {
+      "description": "Represents an acquired phone number.",
+      "required": [
+        "assignmentType",
+        "capabilities",
+        "countryCode",
+        "id",
+        "phoneNumber",
+        "phoneNumberType"
```

Comment:
I think it's better if `callbackUri` and `applicationId` are also required and set to ""?

`purchaseDate` and `cost` should be required, so I'd make everything required.

Reply:
As discussed previously, callbackUri, applicationId will be removed for GA.

Reply:
Also, purchaseDate / cost will be left as optional for now, as we have a bug that these two fields are not populated currently.